### ISSUE TITLE
Nested string interpolation triggers no_unecessary_double_quotes rule

### ISSUE THREAD

**Issue by Blasz** [clutchski/coffeelint#368]
When there is parentheses wrapping nested string interpolation.

```
# coffeelint: enable=no_unnecessary_double_quotes
string = "(#{a("#{b}")})"
```

Removing either the nested string interpolation or either of the two surrounding parentheses does not trigger the rule.

**Comment by AsaAyers** [comment]
I'm not really sure what you're trying to do here, but if you really think you want to use nested interpolation, you'll need to shut off the rule. Just because the syntax compiles doesn't mean it should be accepted by CoffeeLint.

    innerValue = a(b)
    string = "(#{innerValue})"

**Comment by AsaAyers** [comment]
The only thing the inner interpolation is doing is implicitly casting b to a string.

```CoffeeScript
string = "(#{a("#{b}")})"
# string = "(" + (a("" + b)) + ")";

string = "(#{a(b)})"
# string = "(" + (a(b)) + ")";
```

**Comment by Blasz** [comment]
I was providing a minimal test example to show the incorrect linting.Nested interpolation works fine, it's just the parantheses wrapped around it.

A more 'real-world' example would be

```
config = eval("(#{grunt.file.read("#{basePath}/config.js")})")
```
Yes this can be achieved via
```
filePath = "#{basePath}/config.js"
config = eval("(#{grunt.file.read(filePath)})")
```

but I'd prefer to do it in one line.

Does coffeelint have an explicit stance on nested string interpolation?
It seems to be linted correctly with this rule except for the bug stated in this post.

**Comment by AsaAyers** [comment]
I have made all the internals of CoffeeLint available so you can create your own rules. If you really think this is worth having you're welcome to create and use a custom rule.

My stance is that if it makes the code awkward to follow I'm not going to support it. At some point someone is going to need to read that code and there's no reason to force them through the mental hurdles of figuring out your nested interpolation.

**Comment by swang** [comment]
This no longer triggers `no_unecessary_double_quotes` but by default coffeelint will warn against nested string interpolation (`no_nested_string_interpolation`)
